geode-core/src/main/java/org/apache/geode/distributed/internal/InternalLocator.java

```diff
@@ -773,15 +773,19 @@ private void startCache(DistributedSystem system) throws IOException {
     startClusterManagementService();
   }
 
-  private void startClusterManagementService() throws IOException {
+  @VisibleForTesting
+  void startClusterManagementService() throws IOException {
     startConfigurationPersistenceService();
 
-    if (internalCache == null) {
+    InternalCache myCache = this.internalCache;
```

Comment:
don't want to mind-read here, what was the rationale for a local copy?

Reply:
this caches the value to ensure that it doesn't go null after this check